core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -96,6 +96,9 @@ public MetricsReporter metricsReporter() {
     return LoggingMetricsReporter.instance();
   }
 
+  @Nullable
+  public abstract String branch();
```

Comment:
I think it makes more sense to name this `ref()` because the ref can be either a branch or a tag

Reply:
@jackye1995 thoughts on whether this should be named `branch()` vs `ref()`?

Reply:
It can't be a tag, has to be a branch

Reply:
what I mean is: we are expressing that incremental scans work on branches through `IncrementalScan.useBranch()`, which is the publicly exposed API and the correct thing to do. 
My thinking is that `TableScanContext` is rather an internal API and is also used in other places where don't need the distinction between branch/tag. So in order to not limit ourselves in this internal API, I think it makes more sense to name it `ref()` in `TableScanContext`.

Reply:
oh, I know what you mean now, but I have a question. Even though TableScanContext is an internal class, I think it encapsulates the scanning semantics, such as having methods like `useSnapshotId`, `fromSnapshotIdExclusive`, etc. So let’s imagine, if there is a `Tag` here, what would be its use case and how would it relate to Branch. Assuming there is a `fromTagExclusive`, then it would require a Tag, but at the same time, it might also be able to use a Branch. From this perspective, we might need both `tag` and `branch`.

Reply:
Yeah I would +1 on using `branch` instead of a generic `ref`, because here the semantic meaning is quite different. @amogh-jahagirdar any thoughts on that?